### PR TITLE
Add compression for named pipes

### PR DESCRIPTION
**Pull request by bfriesen**

### CODE REVIEW

File: Rock.Messaging/MQ/NamedPipes/NamedPipeReceiverMessage.cs [hunk 1]
```diff
@@ -51,7 +60,7 @@ public byte[] GetBinaryValue(Encoding encoding)
             return
                 stringValue == null
                     ? null
-                    : encoding == null
+                    : MessageFormat == MessageFormat.Binary || encoding == null
```
Comment (bfriesen) [inline Rock.Messaging/MQ/NamedPipes/NamedPipeReceiverMessage.cs]: We should consider performing this check for other messaging implementations.

File: Rock.Messaging/MQ/NamedPipes/NamedPipeMessageSerializer.cs [hunk 2]
```diff
@@ -84,18 +90,18 @@ public object DeserializeFromString(string data, Type type)
 
             Skip(enumerator, _stringValueHeader.Length);
             var stringValue = Unescape(GetStringValue(enumerator));
-            Skip(enumerator, _binaryValueHeader.Length - 1);
-            var binaryValue = Convert.FromBase64String(GetStringValue(enumerator));
             Skip(enumerator, _messageFormatHeader.Length - 1);
             var messageFormat = (MessageFormat)Enum.Parse(typeof(MessageFormat), GetStringValue(enumerator));
+            Skip(enumerator, _priorityHeader.Length);
```
Comment (aydinakcasu) [inline Rock.Messaging/MQ/NamedPipes/NamedPipeMessageSerializer.cs]: should this be _priorityHeader.Length - 1?

Comment (bfriesen) [inline Rock.Messaging/MQ/NamedPipes/NamedPipeMessageSerializer.cs]: Not according the testing I did. (off-by-one errors are hard)

File: Rock.Messaging/MQ/NamedPipes/NamedPipeReceiverMessage.cs (hunk 1 above)
Comment (bfriesen) [inline Rock.Messaging/MQ/NamedPipes/NamedPipeReceiverMessage.cs]: We should consider performing this check for other messaging implementations.

**Comment by mjeaton** [comment]
Reviewed. Looks good.

**Pull request merged**